src/Aop/Pointcut/ClassDeclaresPointcut.php

```diff
@@ -0,0 +1,46 @@
+<?php
+/*
+ * Go! AOP framework
+ *
+ * @copyright Copyright 2016, Lisachenko Alexander <lisachenko.it@gmail.com>
+ *
+ * This source file is subject to the license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Go\Aop\Pointcut;
+
+use Go\Aop\Pointcut;
+use Go\Aop\PointFilter;
+
+/**
+ * Pointcut that filters out all class method and properties which are not declared within class
+ */
+class ClassDeclaresPointcut implements Pointcut
+{
+    use PointcutClassFilterTrait;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function matches($point, $context = null, $instance = null, array $arguments = null)
+    {
+        if (!$context instanceof \ReflectionClass) {
+            return false;
+        };
+
+        if (!($point instanceof \ReflectionMethod) && !($point instanceof \ReflectionProperty)) {
+            return false;
+        }
+
+        return $point->getDeclaringClass()->getName() === $context->getName();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getKind()
+    {
+        return PointFilter::KIND_METHOD | PointFilter::KIND_PROPERTY;
```

Comment:
Yes, you are right, point filter effectively reduce the number of checks for concrete point filter. It's not ideal, but it's very efficient. `KIND_METHOD` filters only methods, `KIND_PROPERTY` filters properties in the class.

For future versions of framework I want to put this logic into `matches()` method. To be in one single place.